ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -803,7 +808,29 @@ public static boolean isPrivate(Map<String, Object> args) throws RestServiceExce
    * @throws RestServiceException if exception occurs during parsing the arg.
    */
   public static boolean isChunkUpload(Map<String, Object> args) throws RestServiceException {
-    return getBooleanHeader(args, Headers.CHUNK_UPLOAD, false);
+    return getBooleanHeader(args, Headers.CHUNK_UPLOAD, false, false);
+  }
+
+  /**
+   * Determine if {@link Headers#NAMED_UPSERT} is set in the request args.
+   * @param args The request arguments.
+   * @return {@code false} if {@link Headers#NAMED_UPSERT} is set to false else return true.
+   * @throws RestServiceException if exception occurs during parsing the arg.
+   */
+  public static boolean isUpsertForNamedBlob(final Map<String, Object> args) throws RestServiceException {
+    return getBooleanHeader(args, Headers.NAMED_UPSERT, false, true);
```

Comment:
So the side effect of assuming the default as true is that even if the user does not pass it, we end up overwriting their old named blob. Is this the agreed upon contract ?

Reply:
Yes, that's what the default to 'upsert' mean.